### PR TITLE
Implement sound VV

### PR DESCRIPTION
**Pull request by metalgearsloth**
No params yet because I'm lazy but paths and collections work.

https://github.com/space-wizards/RobustToolbox/assets/31366439/60fa676f-0d46-43ab-94da-ccf48c823f4d

Video was the WIP version but gives you the idea.

### CODE REVIEW

**Comment by deltanedas** [comment]
vv if rust enum existed :sob:

File: Robust.Client/ViewVariables/ClientViewVariablesManager.cs [hunk 1]
```diff
@@ -222,6 +223,13 @@ public VVPropEditor PropertyFor(Type? type)
                 return new VVPropEditorTimeSpan();
             }
 
+            if (typeof(SoundSpecifier).IsAssignableFrom(type))
+            {
+                var control = new VVPropEditorSoundSpecifier();
+                IoCManager.InjectDependencies(control);
```
Comment (deltanedas) [inline Robust.Client/ViewVariables/ClientViewVariablesManager.cs]: wouldnt it be better to have this in the constructor of the control

Comment (PJB3005) [reply Robust.Client/ViewVariables/ClientViewVariablesManager.cs]: Too late other code already code it.

File: Robust.Server/Audio/AudioSystem.cs [hunk 2]
```diff
@@ -80,6 +80,9 @@ public override (EntityUid Entity, AudioComponent Component)? PlayGlobal(string
     /// <inheritdoc />
     public override (EntityUid Entity, AudioComponent Component)? PlayEntity(string filename, Filter playerFilter, EntityUid uid, bool recordReplay, AudioParams? audioParams = null)
     {
+        if (string.IsNullOrEmpty(filename))
```
Comment (PJB3005) [inline Robust.Server/Audio/AudioSystem.cs:83]: Why these changes?

File: Robust.Client/ViewVariables/Editors/VVPropEditorSoundSpecifier.cs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+        var editBox = new LineEdit()
+        {
+            HorizontalExpand = true,
+            Editable = !ReadOnly,
+        };
+
+        var controls = new BoxContainer()
+        {
+            Orientation = BoxContainer.LayoutOrientation.Vertical,
```
Comment (PJB3005) [inline Robust.Client/ViewVariables/Editors/VVPropEditorSoundSpecifier.cs]: Wouldn't this make way more sense horizontally?

File: Robust.Server/Audio/AudioSystem.cs (hunk 2 above)
Comment (metalgearsloth) [reply Robust.Server/Audio/AudioSystem.cs:83]: SoundSpecifier being passed in as null already does it and this is the functional equivalent. It's mainly if somehow an empty string gets passed in (e.g. datafield or admin input) it doesn't unnecessarily create an audio source that will never be valid.

I could just make it an exception instead?

File: Robust.Client/ViewVariables/Editors/VVPropEditorSoundSpecifier.cs (hunk 3 above)
Comment (metalgearsloth) [reply Robust.Client/ViewVariables/Editors/VVPropEditorSoundSpecifier.cs]: Whenever audioparams readwrite gets added I would've assumed it'd be below but I guess it can probably squeeze in?

Comment (PJB3005) [reply Robust.Client/ViewVariables/Editors/VVPropEditorSoundSpecifier.cs]: I was asking because you have `HorizontalExpand` on the input box which doesn't do anything if it's set to vertical. Whatever though.

File: Robust.Server/Audio/AudioSystem.cs (hunk 2 above)
Comment (PJB3005) [reply Robust.Server/Audio/AudioSystem.cs:83]: If that's the case this API should be made nullable to match the sound specifier API.
